experimental/dds/tree2/src/feature-libraries/editable-tree-2/proxies/types.ts

```diff
@@ -25,6 +25,21 @@ import {
 	Sequence,
 	AssignableFieldKinds,
 } from "../editableTreeTypes";
+import { nodeSym } from "./node";
+
+/**
+ * An object-like SharedTree node.  Includes objects, lists, and maps.
+ * @alpha
+ */
+export interface SharedTreeNode {
```

Comment:
nit: This is technically a "type", but I was expecting to find it in "node.ts", next to the things that are intimately connected to it. But it also makes sense to be here, if you prefer.

Reply:
I've noticed that some projects (and some parts of Fluid) follow a convention of putting everything that is erased by the compiler (types, interfaces, and the right sort of constants) into a single 'types.ts' file, or a 'types.ts' per directory.

I have some theories on why (nostalgia for C/C++ header files, JS projects that incrementally adopted TS by way of *.d.ts typings, ...)

I didn't move SharedTreeNode, but I don't have a real opinion.